src/common/News/NewsCard.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import {
+  View,
+  ActivityIndicator,
+  Text,
+  TouchableHighlight,
+} from 'react-native';
+import { useNavigation } from 'react-navigation-hooks';
+import PropTypes from 'prop-types';
+import { fetchStockNews } from './newsService';
+import styles from '../styles';
+import Card from '../Card';
+import ErrorMessage from '../ErrorMessage';
+import NewsMappedList from './NewsMappedList';
+import CardFooter from '../CardFooter';
+
+const NewsCard = ({ query }) => {
+  const [news, setNews] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState();
+  const navigator = useNavigation();
+  useEffect(() => {
+    setIsLoading(true);
+    setError(undefined);
+    fetchStockNews(query, 1, 2)
```

Comment:
the same as below

Reply:
If you would use redux then you shouldn't load the news in the NewsList component, since you would load them here. However, if you want we can leave it like this, but definitely we will need a custom hook for this